<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0" stdsetdef="1">
  <author></author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>MessageForm</class>
  <widget class="QMainWindow" name="MessageForm">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>578</width>
        <height>356</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Twinkle - Instant message</string>
    </property>
    <widget class="QWidget">
      <layout class="QVBoxLayout">
        <item>
          <layout class="QGridLayout">
            <item row="1" column="0">
              <widget class="QLabel" name="toTextLabel">
                <property name="text">
                  <string>&amp;To:</string>
                </property>
                <property name="buddy" stdset="0">
                  <cstring>toLineEdit</cstring>
                </property>
                <property name="wordWrap">
                  <bool>false</bool>
                </property>
              </widget>
            </item>
            <item row="0" column="1">
              <widget class="QComboBox" name="fromComboBox">
                <property name="whatsThis" stdset="0">
                  <string>The user that will send the message.</string>
                </property>
              </widget>
            </item>
            <item row="1" column="1">
              <layout class="QHBoxLayout">
                <item>
                  <widget class="QLineEdit" name="toLineEdit">
                    <property name="whatsThis" stdset="0">
                      <string>The address of the user that you want to send a message. This can be a full SIP address like &lt;b&gt;sip:example@example.com&lt;/b&gt; or just the user part or telephone number of the full address. When you do not specify a full address, then Twinkle will complete the address by using the domain value of your user profile.</string>
                    </property>
                  </widget>
                </item>
                <item>
                  <widget class="QToolButton" name="addressToolButton">
                    <property name="focusPolicy">
                      <enum>Qt::TabFocus</enum>
                    </property>
                    <property name="text">
                      <string/>
                    </property>
                    <property name="shortcut">
                      <string>F10</string>
                    </property>
                    <property name="icon">
                      <iconset>:/icons/images/kontact_contacts.png</iconset>
                    </property>
                    <property name="toolTip" stdset="0">
                      <string>Address book</string>
                    </property>
                    <property name="whatsThis" stdset="0">
                      <string>Select an address from the address book.</string>
                    </property>
                  </widget>
                </item>
              </layout>
            </item>
            <item row="0" column="0">
              <widget class="QLabel" name="profileTextLabel">
                <property name="text">
                  <string>&amp;User profile:</string>
                </property>
                <property name="buddy" stdset="0">
                  <cstring>fromComboBox</cstring>
                </property>
                <property name="wordWrap">
                  <bool>false</bool>
                </property>
              </widget>
            </item>
          </layout>
        </item>
        <item>
          <widget class="QGroupBox" name="conversationGroupBox">
            <property name="title">
              <string>Conversation</string>
            </property>
            <layout class="QGridLayout">
              <item row="0" column="0">
                <widget class="QTextBrowser" name="conversationBrowser">
                 <property name="sizePolicy">
                  <sizepolicy vsizetype="Expanding" hsizetype="Expanding" />
                 </property>
                <property name="openLinks">
                  <bool>false</bool>
                </property>
                </widget>
              </item>
            </layout>
          </widget>
        </item>
        <item>
          <layout class="QHBoxLayout">
            <item>
              <widget class="QLineEdit" name="msgLineEdit">
                <property name="whatsThis" stdset="0">
                  <string>Type your message here and then press &quot;send&quot; to send it.</string>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QPushButton" name="sendPushButton">
                <property name="text">
                  <string>&amp;Send</string>
                </property>
                <property name="shortcut">
                  <string>Alt+S</string>
                </property>
                <property name="autoDefault">
                  <bool>false</bool>
                </property>
                <property name="default">
                  <bool>true</bool>
                </property>
                <property name="whatsThis" stdset="0">
                  <string>Send the message.</string>
                </property>
              </widget>
            </item>
          </layout>
        </item>
      </layout>
    </widget>
    <widget class="QToolBar" name="Toolbar">
      <addaction name="sendFileAction"/>
    </widget>
    <action name="sendFileAction">
      <property name="icon">
        <iconset>:/icons/images/attach.png</iconset>
      </property>
      <property name="iconText">
        <string>Send file...</string>
      </property>
      <property name="toolTip">
        <string>Send file</string>
      </property>
    </action>
  </widget>
  <layoutdefault spacing="6" margin="11"/>
  <includes>
    <include location="local">ui_getaddressform.h</include>
    <include location="local">qstring.h</include>
    <include location="local">user.h</include>
    <include location="local">im/msg_session.h</include>
    <include location="local">phone.h</include>
    <include location="local">qlabel.h</include>
  </includes>
  <connections>
    <connection>
      <sender>sendPushButton</sender>
      <signal>clicked()</signal>
      <receiver>MessageForm</receiver>
      <slot>sendMessage()</slot>
    </connection>
    <connection>
      <sender>addressToolButton</sender>
      <signal>clicked()</signal>
      <receiver>MessageForm</receiver>
      <slot>showAddressBook()</slot>
    </connection>
    <connection>
      <sender>conversationBrowser</sender>
      <signal>anchorClicked(QUrl)</signal>
      <receiver>MessageForm</receiver>
      <slot>showAttachmentPopupMenu(QUrl)</slot>
    </connection>
    <connection>
      <sender>sendFileAction</sender>
      <signal>triggered()</signal>
      <receiver>MessageForm</receiver>
      <slot>chooseFileToSend()</slot>
    </connection>
    <connection>
      <sender>toLineEdit</sender>
      <signal>textChanged(QString)</signal>
      <receiver>MessageForm</receiver>
      <slot>toAddressChanged(QString)</slot>
    </connection>
    <connection>
      <sender>msgLineEdit</sender>
      <signal>textChanged(QString)</signal>
      <receiver>MessageForm</receiver>
      <slot>showMessageSize()</slot>
    </connection>
  </connections>
</ui>
